File: Backend/src/main/java/com/example/backend/CafeteriaStaff.java
package com.example.backend;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CafeteriaStaff extends User{

    public CafeteriaStaff(String name, String mail, String password, String role,String address, String phoneNumber, String hesCode){
        super(name,mail,password,role,address,phoneNumber,hesCode);
    }

    //Methods
    public String showTotalInfo(){
        return null;
    }
    public String ShowRegInfo(String region){
        return null;
    }
    public Boolean searchIndividualInfo(int ID){
        return null;
    }
    public void enterTakenStatus(Meal meal){

    }
}

File: Frontend/src/classes/CafeteriaStaff.ts
import User from './User'
import Meal from './Meal'
import CafeteriaManager from './CafeteriaManager'

export default class CafeteriaStaff extends User {
  
  //Constructor
  public constructor(name: string, mail: string, password: string, role: string, address: string, phoneNumber: string, hesCode: string) {
    super(name,mail,password,role,address,phoneNumber,hesCode);
  }

  //Methods
  public showTotalInfo(){
    let cm = CafeteriaManager.getInstance();
    cm.seeTotalOrders();
  }

  public ShowRegInfo(region: string){
    let cm = CafeteriaManager.getInstance();
    cm.seeRegionalOrders(region);
  }

  public enterTakenStatus(UID:string, OID: string){
    let cm = CafeteriaManager.getInstance();
    cm.changeTakenStatus(UID, OID);
  }

}